File: src/main/java/com/example/culturaleventssystem/service/FavoriteService.java
package com.example.culturaleventssystem.service;

import com.example.culturaleventssystem.entity.Favorite;
import com.example.culturaleventssystem.repository.FavoriteRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class FavoriteService {

    @Autowired
    private FavoriteRepository favoriteRepository;

    // Добавить в избранное
    public Favorite addToFavorites(Favorite favorite) {
        // Проверяем, не добавлено ли уже в избранное
        if (favoriteRepository.existsByUserIdAndEventId(
                favorite.getUser().getId(), favorite.getEvent().getId())) {
            throw new RuntimeException("Event is already in favorites");
        }
        return favoriteRepository.save(favorite);
    }

    // Удалить из избранного
    public void removeFromFavorites(Long userId, Long eventId) {
        Favorite favorite = favoriteRepository.findByUserIdAndEventId(userId, eventId)
                .orElseThrow(() -> new RuntimeException("Favorite not found"));
        favoriteRepository.delete(favorite);
    }

    // Получить все избранные мероприятия пользователя
    public List<Favorite> getUserFavorites(Long userId) {
        return favoriteRepository.findByUserId(userId);
    }

    // Проверить, есть ли мероприятие в избранном у пользователя
    public boolean isEventInFavorites(Long userId, Long eventId) {
        return favoriteRepository.existsByUserIdAndEventId(userId, eventId);
    }

    // Получить количество избранных мероприятий у пользователя
    public Long getUserFavoriteCount(Long userId) {
        return favoriteRepository.countByUserId(userId);
    }

    // Найти запись избранного
    public Optional<Favorite> getFavoriteByUserAndEvent(Long userId, Long eventId) {
        return favoriteRepository.findByUserIdAndEventId(userId, eventId);
    }

    // Получить все избранные
    public List<Favorite> getAllFavorites() {
        return favoriteRepository.findAll();
    }
}

File: frontend/src/services/favoriteService.js
import api from './api';

export const favoriteService = {
  // Получить все избранные пользователя
  getUserFavorites: async (userId) => {
    const response = await api.get(`/favorites/user/${userId}`);
    return response.data;
  },

  // Добавить в избранное
  addToFavorites: async (favoriteData) => {
    const response = await api.post('/favorites', favoriteData);
    return response.data;
  },

  // Удалить из избранного
  removeFromFavorites: async (userId, eventId) => {
    await api.delete(`/favorites/user/${userId}/event/${eventId}`);
  },

  // Проверить, есть ли в избранном
  isEventInFavorites: async (userId, eventId) => {
    const response = await api.get(`/favorites/user/${userId}/event/${eventId}/check`);
    return response.data.isFavorite;
  },

  // Получить количество избранных
  getUserFavoriteCount: async (userId) => {
    const response = await api.get(`/favorites/user/${userId}/count`);
    return response.data.favoriteCount;
  }
};